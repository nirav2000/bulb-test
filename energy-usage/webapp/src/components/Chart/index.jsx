import React from 'react';
import { BarChart, Bar, CartesianGrid, XAxis, YAxis, Tooltip } from 'recharts';
import PropTypes from 'prop-types';

const Chart = ({ energyUsageData }) => (
  <div>
    <h2>Energy Usage</h2>
    <BarChart width={1400} height={400} data={energyUsageData}>
      <XAxis dataKey="date" />
      <YAxis dataKey="energyUsage" />
      <CartesianGrid horizontal={false} />
      <Tooltip />
      <Bar
        dataKey="energyUsage"
        fill="#03ad54"
        isAnimationActive={false}
      />
    </BarChart>
  </div>
);

Chart.propTypes = {
  energyUsageData: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Chart;
