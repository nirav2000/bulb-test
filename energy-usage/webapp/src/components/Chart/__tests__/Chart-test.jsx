import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import data from '../../data/../../data/meterReadingsSample.json';
import Chart from '../index';

describe('<Chart /> test', () => {
  it('Test snapshots of content', () => {
    const wrapper = shallow(<Chart energyUsageData={data.electricity} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
