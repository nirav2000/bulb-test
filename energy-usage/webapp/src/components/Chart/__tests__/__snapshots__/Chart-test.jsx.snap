// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart /> test Test snapshots of content 1`] = `
<div>
  <h2>
    Energy Usage
  </h2>
  <BarChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "cumulative": 17600,
          "readingDate": "2017-03-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 17859,
          "readingDate": "2017-04-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18102,
          "readingDate": "2017-05-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18290,
          "readingDate": "2017-06-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18453,
          "readingDate": "2017-07-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18620,
          "readingDate": "2017-08-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18782,
          "readingDate": "2017-09-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18965,
          "readingDate": "2017-10-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 19230,
          "readingDate": "2017-11-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 19517,
          "readingDate": "2017-12-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 19827,
          "readingDate": "2018-01-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 20113,
          "readingDate": "2018-02-28T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 20376,
          "readingDate": "2018-03-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 20610,
          "readingDate": "2018-04-30T00:00:00.000Z",
          "unit": "kWh",
        },
      ]
    }
    height={400}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
    width={1400}
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="date"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="energyUsage"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="number"
      width={60}
      yAxisId={0}
    />
    <CartesianGrid
      fill="none"
      horizontal={false}
      horizontalFill={Array []}
      horizontalPoints={Array []}
      stroke="#ccc"
      vertical={true}
      verticalFill={Array []}
      verticalPoints={Array []}
    />
    <Tooltip
      active={false}
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemSorter={[Function]}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <Bar
      animationBegin={0}
      animationDuration={400}
      animationEasing="ease"
      data={Array []}
      dataKey="energyUsage"
      fill="#03ad54"
      hide={false}
      isAnimationActive={false}
      layout="vertical"
      legendType="rect"
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
  </BarChart>
</div>
`;
